surrealdb/clients/http.py

```diff
@@ -0,0 +1,169 @@
+"""
+Copyright © SurrealDB Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from json import JSONDecodeError
+from types import TracebackType
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Type
+
+import httpx
+
+from ..common import json as jsonlib
+from ..common.exceptions import SurrealException
+from ..models.response import SurrealResponse
+
+__all__ = ("SurrealDBHTTPClient",)
+
+
+class SurrealDBHTTPClient:
+    def __init__(
+        self,
+        url: str,
+        *,
+        namespace: str,
+        database: str,
+        username: str,
+        password: str,
+    ) -> None:
+        self._url = url
+        self._namespace = namespace
+        self._database = database
+        self._username = username
+        self._password = password
+
+        self._http = httpx.AsyncClient(
+            base_url=self._url,
+            auth=httpx.BasicAuth(
+                username=self._username,
+                password=self._username,
+            ),
+            headers={
+                "NS": self._namespace,
+                "DB": self._database,
+                "Accept": "application/json",
+                "Content-Type": "application/json",
+            },
+        )
+
+    async def __aenter__(self) -> "SurrealDBHTTPClient":
```

Comment:
If you include;
```py
from __future__ import annotations
```
You don't need to encase this import in quotes

```suggestion
    async def __aenter__(self) -> SurrealDBHTTPClient:
```

Reply:
forgot oops